pkg/knowledge_base2/properties/any_property.go

```diff
@@ -107,7 +109,10 @@ func (a *AnyProperty) Type() string {
 	return "any"
 }
 
-func (a *AnyProperty) Validate(resource *construct.Resource, value any) error {
+func (a *AnyProperty) Validate(resource *construct.Resource, value any, ctx knowledgebase.DynamicContext) error {
+	if a.Required && value == nil {
```

Comment:
Considering the infracopilot UI flow, it might not be possible to do required checks until we figure that out (unless that's something you're already doing). In other words, the resource has to be present to configure it in the UI, so if the validation fails on adding the resource, the user can't configure it to resolve that validation.

Reply:
yeah validation can fail, but the engine wont fail unless its correlated with an incoming resource constraint. This prevents the user from dragging in a lambda and the engine failing, but still gives us the validation info for the images BaseImage so we can block clicking export IAC until all fields pass